### DISCUSSION TITLE
QR code in emails

### DISCUSSION THREAD

**Discussion by sahilkanojia494** [Faire/mjml-react#127] [Q&A] [answered]
Hi Team,
Requirement: I have a requirement to put QR code in emails for redirecting the customers from web to app.

I was looking the possible solutions but didn't find any solid solution.

I came to this [post](https://documentation.mjml.io/#mj-qr-code) where they mentioned about [mjml-qr-code](https://www.npmjs.com/package/mjml-qr-code) npm package. But it seems this library is not built around the react which makes hard to use it in my react+nodejs project.

if you have other solutions then let me know I'm happy to explore more solutions.

Thank you.

**Comment by Bertg** [comment, marked as answer]
I would not delegate this to Mjml. 
Just generate the QR-code as an image (I'm sure you can find a suitable url based service) and embed that image in your email. 

This way you are sure that the image will always be rendered correctly, and if you control the img generating service, you can even update the codes after the email is sent.
